contracts/SwapKiwi.sol

```diff
@@ -2,61 +2,70 @@
 
 pragma solidity ^0.8.1;
 
-import "@openzeppelin/contracts/token/ERC721/IERC721Receiver.sol";
+import "@openzeppelin/contracts/token/ERC721/utils/ERC721Holder.sol";
+import "@openzeppelin/contracts/token/ERC1155/utils/ERC1155Holder.sol";
 import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
 import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC1155/IERC1155.sol";
 import "@openzeppelin/contracts/access/Ownable.sol";
 
-contract SwapKiwi is Ownable, IERC721Receiver {
+/**
+* @title This is the contract which added erc1155 into the previous swap contract.
+*/
+contract SwapKiwi is Ownable, ERC721Holder, ERC1155Holder {
 
-  uint256 private _swapsCounter;
-  uint256 private _etherLocked;
+  uint64 private _swapsCounter;
+  uint128 private _etherLocked;
+  uint128 public fee;
 
-  uint256 public fee;
-
-  mapping (uint256 => Swap) private _swaps;
+  mapping (uint64 => Swap) private _swaps;
 
   struct Swap {
     address payable initiator;
     address[] initiatorNftAddresses;
     uint256[] initiatorNftIds;
-    uint256 initiatorEtherValue;
+    uint256[] initiatorNftAmounts;
```

Comment:
We would merely require to include it multiple times (the same nft address, same nft id, and total split into a bunch of amounts), but in 99.99% of cases it will save gas when more than 1 nftAddress is used.